sympy/core/mul.py

```diff
@@ -2112,7 +2112,7 @@ def _keep_coeff(coeff, factors, clear=True, sign=False):
         if not clear and coeff.is_Rational and coeff.q != 1:
             args = [i.as_coeff_Mul() for i in factors.args]
             args = [(_keep_coeff(c, coeff), m) for c, m in args]
-            if any(c == int(c) for c, _ in args):
+            if any(not c.is_infinite and c == int(c) for c, _ in args):
```

Comment:
The condition should probably be something like `c.is_Number` as `int(c)` is not implemented for finite complex numbers:
```
In [23]: I.is_infinite                                                          
Out[23]: False

In [24]: int(I)                                                                 
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)

TypeError: can't convert complex to int

In [25]: I.is_Number                                                            
Out[25]: False
```

Reply:
Ahh, I only noted the `oo` error, but that clearly makes sense. (For some reason the code doesn't come here for e.g. `oo + I`, only for `oo+I/2` etc, and apparently not for many other cases with complex numbers...)

Reply:
However, as `oo` is a Number, I am not sure exactly what it should be. Right now it works and it seems like `c` rarely (if ever) an imaginary/complex number.

Also, `c` should by sympified as far as I can tell, so not obvious what the purpose of `c == int(c)` is? Check that `c` is an integer or something else? Maybe `is_Integer` is enough?

Reply:
I tried `is_Integer` only and so far so good. Pushing to run the complete test set.

Reply:
> Also, `c` should by sympified as far as I can tell, so not obvious what the purpose of `c == int(c)` is

Yes, `c` was generated from `as_coeff_Mul` so it will be sympified. Even I am not sure why the `int` was used.